### PR TITLE
added stack class

### PR DESCRIPTION
**Pull request by marcelluethi**

### CODE REVIEW

File: Queue.java [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
+		Queue(int size) {
+				this.size = size;
+				q = new int[size];
+		}
+
+		void put(int x) {
+				if ((tail + 1) % size == head) {
+						overflow = true;
+				} else {
+						overflow = true;
```
Comment (swissmathmodel) [inline Queue.java:20]: Overflow is then always true, even when it does not pass if-condition

File: Queue.java [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+				}
+
+				if (overflow == true) {
+					q[tail]  =x;
```
Comment (swissmathmodel) [inline Queue.java:24]: White spaces displaced - only one white space after q[tail], and one after = so that the line reads q[tail] = x;

**Comment by LuVogel** [comment]
#4 No documentation available `@@ -1,40 +1,40 @@`

**Comment by LuVogel** [comment]
`  void get() ` : Methode ist mit void deklariert hat jedoch return-Statement. Widerspruch!

File: Queue.java [hunk 3]
```diff
@@ -1,40 +1,40 @@
...
+		}
+
+		void put(int x) {
+				if ((tail + 1) % size == head) {
+						overflow = true;
+				} else {
+						overflow = true;
+				}
+
+				if (overflow == true) {
```
Comment (swissmathmodel) [inline Queue.java:23]: Can be simplified to if (overflow){} as boolean

File: Queue.java [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
-    void get() {
-        underflow = head = tail;
-        if (underflow) return -1;
-        else {
-            int x = q[head];
-            head = (head + 1)% size;
-            return x;
-        }
-    }
+		int q[];
```
Comment (swissmathmodel) [inline Queue.java:3]: Not Java declaration of int array -- Java declaration follows int[] q

**Comment by LuVogel** [comment]
` underflow = head = tail;` no valid operation. underflow = head == tail?

File: Queue.java [hunk 5]
```diff
@@ -1,40 +1,40 @@
...
+						overflow = true;
+				}
+
+				if (overflow == true) {
+					q[tail]  =x;
+					tail = (++tail) % size;
+				}
+		}
+
+		void get() {
```
Comment (swissmathmodel) [inline Queue.java:29]: Invalid method type - void does not accept returns

File: Queue.java [hunk 6]
```diff
@@ -1,40 +1,40 @@
...
+				underflow = head = tail;
```
Comment (swissmathmodel) [inline Queue.java:30]: Boolean cannot be equal to an int

File: Stack.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            overflow = true;
+        }
+        else {
+            overflow = false;
+        }
+      if (overflow == true)  s[length++] = x;
+    }
+
+      int pop() {
+        underflow = lenght == 0;
```
Comment (swissmathmodel) [inline Stack.java:27]: Typo here with length (spelled lenght)

File: Stack.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    void push(int x)   {
+        if (length >= s.length) {
+            overflow = true;
+        }
+        else {
+            overflow = false;
+        }
+      if (overflow == true)  s[length++] = x;
```
Comment (swissmathmodel) [inline Stack.java:23]: Can be simplified to just if (overflow){} as overflow is boolean

**Review by swissmathmodel — COMMENTED** [review]
Some formatting and Java errors, see above

**Pull request closed without merging**